service_manager.war cannot start

`setup.sh` successfully installs all required software  as well as deploys tomcat instance. However `service_manger.war` webapp does not start up. 
It seems that for some reason postgres jdbc cannot pass through postgres authentication. 
`org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"`

Attached full catalina log. 
```
OpenJDK 64-Bit Server VM warning: ignoring option PermSize=256m; support was removed in 8.0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=1024m; support was removed in 8.0
Feb 18, 2017 11:03:10 AM org.apache.catalina.core.AprLifecycleListener init
INFO: The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
Feb 18, 2017 11:03:10 AM org.apache.coyote.http11.Http11Protocol init
INFO: Initializing Coyote HTTP/1.1 on http-8080
Feb 18, 2017 11:03:10 AM org.apache.catalina.startup.Catalina load
INFO: Initialization processed in 347 ms
Feb 18, 2017 11:03:10 AM org.apache.catalina.core.StandardService start
INFO: Starting service Catalina
Feb 18, 2017 11:03:10 AM org.apache.catalina.core.StandardEngine start
INFO: Starting Servlet Engine: Apache Tomcat/6.0.35
Feb 18, 2017 11:03:10 AM org.apache.catalina.startup.HostConfig deployDescriptor
INFO: Deploying configuration descriptor host-manager.xml
Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
INFO: Deploying configuration descriptor jp.go.nict.langrid.webapps.blank.xml
Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
INFO: Deploying configuration descriptor manager.xml
Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
INFO: Deploying configuration descriptor SampleTranslationService.xml
Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
INFO: Deploying configuration descriptor service_manager.xml
Feb 18, 2017 11:03:12 AM jp.go.nict.langrid.servlet.filter.auth.ApplicationAuthenticator init
WARNING: RichAppAuth: parameter loading failed.failed to initialize filter: 
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.Version <clinit>
INFO: Hibernate Annotations 3.4.0.GA
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
INFO: Hibernate 3.3.2.GA
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
INFO: hibernate.properties not found
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment buildBytecodeProvider
INFO: Bytecode provider name : javassist
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
INFO: using JDK 1.4 java.sql.Timestamp handling
Feb 18, 2017 11:03:12 AM org.hibernate.annotations.common.Version <clinit>
INFO: Hibernate Commons Annotations 3.1.0.GA
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration configure
INFO: configuring from url: file:/usr/share/tomcat/service-manager/webapps/service_manager/WEB-INF/classes/hibernate.cfg.xml
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration doConfigure
INFO: Configured SessionFactory: null
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration configure
INFO: configuring from url: file:/usr/share/tomcat/service-manager/webapps/service_manager/WEB-INF/classes/hibernate.cfg.xml
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration doConfigure
INFO: Configured SessionFactory: null
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.search.HibernateSearchEventListenerRegister enableHibernateSearch
INFO: Unable to find org.hibernate.search.event.FullTextIndexEventListener on the classpath. Hibernate Search is not enabled.
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.SystemProperty
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.SystemProperty on table SystemProperty
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Grid
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Grid on table Grid
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.GridAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.GridAttribute on table GridAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Federation
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Federation on table Federation
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.User
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.User on table users
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.UserAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.UserAttribute on table UserAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.UserRole
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.UserRole on table userRoles
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.TemporaryUser
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.QueryBinder bindQuery
INFO: Binding Named query: TemporaryUser.isParent => select count(*) from TemporaryUser where gridId=:gridId and parentUserId=:parentUserId and userId=:userId
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.TemporaryUser on table TemporaryUser
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Resource
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Resource on table Resource
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceAttribute on table ResourceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Service
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Service on table Service
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceAttribute on table ServiceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceDeployment
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceDeployment on table ServiceDeployment
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceEndpoint
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceEndpoint on table ServiceEndpoint
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ExternalService
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.BPELService
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.WebappService
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Invocation
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Invocation on table Invocation
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Node
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Node on table Node
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.NodeAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.NodeAttribute on table NodeAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessLog
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessLog on table AccessLog
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessLimit
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessLimit on table AccessLimit
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessRight
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessRight on table AccessRight
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessStat
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.QueryBinder bindQuery
INFO: Binding Named query: AccessStat.incrementQuery => update AccessStat set  accesscount = accesscount + 1 , transferredsize = transferredsize + :size where userId=:userId  and serviceId=:serviceId and nodeId=:nodeId and (   (period=:periodDay and baseDateTime=:dateTimeDay)   or   (period=:periodMonth and baseDateTime=:dateTimeMonth)   or   (period=:periodYear and baseDateTime=:dateTimeYear)   )
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessStat on table AccessStat
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.OverUseLimit
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.OverUseLimit on table OverUseLimit
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AcceptableRemoteAddress
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.AcceptableRemoteAddress on table AcceptableRemoteAddress
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.OperationRequest
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.OperationRequest on table OperationRequest
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceActionSchedule
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceActionSchedule on table ServiceActionSchedule
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.News
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.News on table News
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Domain
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Domain on table Domain
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Protocol
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.Protocol on table Protocol
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceType
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceType on table ResourceType
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceMetaAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceMetaAttribute on table ResourceMetaAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceType
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceType on table ServiceType
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceMetaAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceMetaAttribute on table ServiceMetaAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceInterfaceDefinition
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceInterfaceDefinition on table ServiceInterfaceDefinition
Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
INFO: Using dialect: org.hibernate.dialect.HSQLDialect
Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
INFO: Using dialect: org.hibernate.dialect.HSQLDialect
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Grid.attributes -> GridAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.User.attributes -> UserAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.User.roles -> userRoles
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Resource.attributes -> ResourceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.attributes -> ServiceAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.deployments -> ServiceDeployment
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.endpoints -> ServiceEndpoint
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.invocations -> Invocation
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Node.attributes -> NodeAttribute
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
INFO: Mapping collection: jp.go.nict.langrid.dao.entity.ServiceType.interfaceDefinitions -> ServiceInterfaceDefinition
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationConfiguration secondPassCompile
INFO: Hibernate Validator not found: ignoring
Feb 18, 2017 11:03:12 AM org.hibernate.util.NamingHelper getInitialContext
INFO: JNDI InitialContext properties:{}
Feb 18, 2017 11:03:12 AM org.hibernate.connection.DatasourceConnectionProvider configure
INFO: Using datasource: java:comp/env/jdbc/langrid
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
WARNING: Could not obtain connection to query metadata
org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
	at org.hibernate.cfg.SettingsFactory.buildSettings(SettingsFactory.java:111)
	at org.hibernate.cfg.Configuration.buildSettingsInternal(Configuration.java:2119)
	at org.hibernate.cfg.Configuration.buildSettings(Configuration.java:2115)
	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1339)
	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
	at org.postgresql.Driver.makeConnection(Driver.java:391)
	at org.postgresql.Driver.connect(Driver.java:265)
	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
	... 50 more

Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
INFO: Using dialect: org.hibernate.dialect.PostgreSQLDialect
Feb 18, 2017 11:03:12 AM org.hibernate.transaction.TransactionFactoryFactory buildTransactionFactory
INFO: Using default transaction strategy (direct JDBC transactions)
Feb 18, 2017 11:03:12 AM org.hibernate.transaction.TransactionManagerLookupFactory getTransactionManagerLookup
INFO: No TransactionManagerLookup configured (in JTA environment, use of read-write or transactional second-level cache is not recommended)
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Automatic flush during beforeCompletion(): disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Automatic session close at end of transaction: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Scrollable result sets: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: JDBC3 getGeneratedKeys(): disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Connection release mode: auto
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Default batch fetch size: 1
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Generate SQL with comments: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Order SQL updates by primary key: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Order SQL inserts for batching: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory createQueryTranslatorFactory
INFO: Query translator: org.hibernate.hql.ast.ASTQueryTranslatorFactory
Feb 18, 2017 11:03:12 AM org.hibernate.hql.ast.ASTQueryTranslatorFactory <init>
INFO: Using ASTQueryTranslatorFactory
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Query language substitutions: {}
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: JPA-QL strict compliance: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Second-level cache: enabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Query cache: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory createRegionFactory
INFO: Cache region factory : org.hibernate.cache.impl.bridge.RegionFactoryCacheProviderBridge
Feb 18, 2017 11:03:12 AM org.hibernate.cache.impl.bridge.RegionFactoryCacheProviderBridge <init>
INFO: Cache provider: org.hibernate.cache.NoCacheProvider
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Optimize cache for minimal puts: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Structured second-level cache entries: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Statistics: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Deleted entity synthetic identifier rollback: disabled
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Default entity-mode: pojo
Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
INFO: Named query checking : enabled
Feb 18, 2017 11:03:12 AM org.hibernate.impl.SessionFactoryImpl <init>
INFO: building session factory
Feb 18, 2017 11:03:12 AM org.hibernate.impl.SessionFactoryObjectFactory addInstance
INFO: Not binding factory to JNDI, no JNDI name configured
Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
INFO: Running hbm2ddl schema update
Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
INFO: fetching database metadata
Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
SEVERE: could not get database metadata
org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
	at org.hibernate.tool.hbm2ddl.SuppliedConnectionProviderConnectionHelper.prepare(SuppliedConnectionProviderConnectionHelper.java:51)
	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:168)
	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:356)
	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1341)
	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
	at org.postgresql.Driver.makeConnection(Driver.java:391)
	at org.postgresql.Driver.connect(Driver.java:265)
	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
	... 50 more

Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
SEVERE: could not complete schema update
org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
	at org.hibernate.tool.hbm2ddl.SuppliedConnectionProviderConnectionHelper.prepare(SuppliedConnectionProviderConnectionHelper.java:51)
	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:168)
	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:356)
	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1341)
	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
	at org.postgresql.Driver.makeConnection(Driver.java:391)
	at org.postgresql.Driver.connect(Driver.java:265)
	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
	... 50 more

Feb 18, 2017 11:03:13 AM org.hibernate.dialect.Dialect <init>
INFO: Using dialect: org.hibernate.dialect.PostgreSQLDialect
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
WARNING: SQL Error: 0, SQLState: null
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
11:03:13,119 ERROR ServiceManagerApplication:124 - [TreadID:1,UserID:Spring Transaction,ClassName:ExtendTransactionInterceptor]
jp.go.nict.langrid.dao.ConnectException: org.hibernate.exception.GenericJDBCException: Cannot open connection
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:151)
	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
Caused by: org.hibernate.exception.GenericJDBCException: Cannot open connection
	at org.hibernate.exception.SQLStateConverter.handledNonSpecificException(SQLStateConverter.java:126)
	at org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:114)
	at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:66)
	at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:52)
	at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:449)
	at org.hibernate.jdbc.ConnectionManager.getConnection(ConnectionManager.java:167)
	at org.hibernate.jdbc.JDBCContext.connection(JDBCContext.java:142)
	at org.hibernate.transaction.JDBCTransaction.begin(JDBCTransaction.java:85)
	at org.hibernate.impl.SessionImpl.beginTransaction(SessionImpl.java:1354)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.hibernate.context.ThreadLocalSessionContext$TransactionProtectionWrapper.invoke(ThreadLocalSessionContext.java:342)
	at com.sun.proxy.$Proxy73.beginTransaction(Unknown Source)
	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:147)
	... 39 more
Caused by: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
	at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:446)
	... 50 more
Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
	at org.postgresql.Driver.makeConnection(Driver.java:391)
	at org.postgresql.Driver.connect(Driver.java:265)
	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
	... 54 more
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
WARNING: SQL Error: 0, SQLState: null
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
WARNING: SQL Error: 0, SQLState: null
Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
Feb 18, 2017 11:03:13 AM org.apache.catalina.core.StandardContext start
SEVERE: Error filterStart
Feb 18, 2017 11:03:13 AM org.apache.catalina.core.StandardContext start
SEVERE: Context [/service_manager] startup failed due to previous errors
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-1] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-2] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-3] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-4] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-5] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-6] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-7] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-8] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-9] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-10] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [Timer-0] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
SEVERE: The web application [/service_manager] appears to have started a thread named [PageSavingThread-ServiceManagerFilter] but has failed to stop it. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearThreadLocalMap
SEVERE: The web application [/service_manager] created a ThreadLocal with key of type [jp.go.nict.langrid.dao.hibernate.HibernateDaoContext$1] (value [jp.go.nict.langrid.dao.hibernate.HibernateDaoContext$1@779de014]) and a value of type [java.lang.Integer] (value [0]) but failed to remove it when the web application was stopped. This is very likely to create a memory leak.
Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
INFO: registered listener interface [RequestListenerInterface name=ILinkListener, method=public abstract void org.apache.wicket.markup.html.link.ILinkListener.onLinkClicked()]
Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
INFO: registered listener interface [RequestListenerInterface name=INewBrowserWindowListener, method=public abstract void org.apache.wicket.markup.html.INewBrowserWindowListener.onNewBrowserWindow()]
Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
INFO: registered listener interface [RequestListenerInterface name=IResourceListener, method=public abstract void org.apache.wicket.IResourceListener.onResourceRequested()]
Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
INFO: registered listener interface [RequestListenerInterface name=IRedirectListener, method=public abstract void org.apache.wicket.IRedirectListener.onRedirect()]
Feb 18, 2017 11:03:13 AM org.apache.catalina.startup.HostConfig deployDirectory
INFO: Deploying web application directory ROOT
Feb 18, 2017 11:03:13 AM org.apache.coyote.http11.Http11Protocol start
INFO: Starting Coyote HTTP/1.1 on http-8080
Feb 18, 2017 11:03:13 AM org.apache.jk.common.ChannelSocket init
INFO: JK: ajp13 listening on /0.0.0.0:8009
Feb 18, 2017 11:03:13 AM org.apache.jk.server.JkMain start
INFO: Jk running ID=0 time=0/8  config=null
Feb 18, 2017 11:03:13 AM org.apache.catalina.startup.Catalina start
INFO: Server startup in 2772 ms
Feb 18, 2017 11:03:24 AM org.apache.catalina.startup.HostConfig checkResources
INFO: Undeploying context [/host-manager]
Feb 18, 2017 11:03:24 AM org.apache.catalina.startup.HostConfig checkResources
INFO: Undeploying context [/manager]
```

I was just looking into this.  I can connect with psql if I edit the file /var/lib/pgsql/9.6/data/pg_hba.conf file and change the METHOD to "trust" for each entry. i.e.

```bash
# "local" is for Unix domain socket connections only                                                                                                   
local   all             all                                     trust
# IPv4 local connections:                                                                                                                              
host    all             all             127.0.0.1/32		trust
# IPv6 local connections:                                                                                                                              
host    all             all             ::1/128			trust
```

I'm doing a Ubuntu install to compare the Postgres setups.

> On Feb 18, 2017, at 11:11 AM, Keigh Rim <notifications@github.com> wrote:
> 
> setup.sh successfully installs all required software as well as deploys tomcat instance. However service_manger.war webapp does not start up.
> It seems that for some reason postgres jdbc cannot pass through postgres authentication.
> org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
> 
> Attached full catalina log.
> 
> OpenJDK 64-Bit Server VM warning: ignoring option PermSize=256m; support was removed in 8.0
> OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=1024m; support was removed in 8.0
> Feb 18, 2017 11:03:10 AM org.apache.catalina.core.AprLifecycleListener init
> INFO: The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
> Feb 18, 2017 11:03:10 AM org.apache.coyote.http11.Http11Protocol init
> INFO: Initializing Coyote HTTP/1.1 on http-8080
> Feb 18, 2017 11:03:10 AM org.apache.catalina.startup.Catalina load
> INFO: Initialization processed in 347 ms
> Feb 18, 2017 11:03:10 AM org.apache.catalina.core.StandardService start
> INFO: Starting service Catalina
> Feb 18, 2017 11:03:10 AM org.apache.catalina.core.StandardEngine start
> INFO: Starting Servlet Engine: Apache Tomcat/6.0.35
> Feb 18, 2017 11:03:10 AM org.apache.catalina.startup.HostConfig deployDescriptor
> INFO: Deploying configuration descriptor host-manager.xml
> Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
> INFO: Deploying configuration descriptor jp.go.nict.langrid.webapps.blank.xml
> Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
> INFO: Deploying configuration descriptor manager.xml
> Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
> INFO: Deploying configuration descriptor SampleTranslationService.xml
> Feb 18, 2017 11:03:11 AM org.apache.catalina.startup.HostConfig deployDescriptor
> INFO: Deploying configuration descriptor service_manager.xml
> Feb 18, 2017 11:03:12 AM jp.go.nict.langrid.servlet.filter.auth.ApplicationAuthenticator init
> WARNING: RichAppAuth: parameter loading failed.failed to initialize filter: 
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.Version <clinit>
> INFO: Hibernate Annotations 3.4.0.GA
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
> INFO: Hibernate 3.3.2.GA
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
> INFO: hibernate.properties not found
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment buildBytecodeProvider
> INFO: Bytecode provider name : javassist
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Environment <clinit>
> INFO: using JDK 1.4 java.sql.Timestamp handling
> Feb 18, 2017 11:03:12 AM org.hibernate.annotations.common.Version <clinit>
> INFO: Hibernate Commons Annotations 3.1.0.GA
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration configure
> INFO: configuring from url: file:/usr/share/tomcat/service-manager/webapps/service_manager/WEB-INF/classes/hibernate.cfg.xml
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration doConfigure
> INFO: Configured SessionFactory: null
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration configure
> INFO: configuring from url: file:/usr/share/tomcat/service-manager/webapps/service_manager/WEB-INF/classes/hibernate.cfg.xml
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.Configuration doConfigure
> INFO: Configured SessionFactory: null
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.search.HibernateSearchEventListenerRegister enableHibernateSearch
> INFO: Unable to find org.hibernate.search.event.FullTextIndexEventListener on the classpath. Hibernate Search is not enabled.
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.SystemProperty
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.SystemProperty on table SystemProperty
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Grid
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Grid on table Grid
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.GridAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.GridAttribute on table GridAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Federation
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Federation on table Federation
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.User
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.User on table users
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.UserAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.UserAttribute on table UserAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.UserRole
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.UserRole on table userRoles
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.TemporaryUser
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.QueryBinder bindQuery
> INFO: Binding Named query: TemporaryUser.isParent => select count(*) from TemporaryUser where gridId=:gridId and parentUserId=:parentUserId and userId=:userId
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.TemporaryUser on table TemporaryUser
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Resource
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Resource on table Resource
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceAttribute on table ResourceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Service
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Service on table Service
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceAttribute on table ServiceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceDeployment
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceDeployment on table ServiceDeployment
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceEndpoint
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceEndpoint on table ServiceEndpoint
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ExternalService
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.BPELService
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.WebappService
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Invocation
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Invocation on table Invocation
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Node
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Node on table Node
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.NodeAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.NodeAttribute on table NodeAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessLog
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessLog on table AccessLog
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessLimit
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessLimit on table AccessLimit
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessRight
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessRight on table AccessRight
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AccessStat
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.QueryBinder bindQuery
> INFO: Binding Named query: AccessStat.incrementQuery => update AccessStat set  accesscount = accesscount + 1 , transferredsize = transferredsize + :size where userId=:userId  and serviceId=:serviceId and nodeId=:nodeId and (   (period=:periodDay and baseDateTime=:dateTimeDay)   or   (period=:periodMonth and baseDateTime=:dateTimeMonth)   or   (period=:periodYear and baseDateTime=:dateTimeYear)   )
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.AccessStat on table AccessStat
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.OverUseLimit
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.OverUseLimit on table OverUseLimit
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.AcceptableRemoteAddress
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.AcceptableRemoteAddress on table AcceptableRemoteAddress
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.OperationRequest
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.OperationRequest on table OperationRequest
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceActionSchedule
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceActionSchedule on table ServiceActionSchedule
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.News
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.News on table News
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Domain
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Domain on table Domain
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.Protocol
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.Protocol on table Protocol
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceType
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceType on table ResourceType
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ResourceMetaAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ResourceMetaAttribute on table ResourceMetaAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceType
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceType on table ServiceType
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceMetaAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceMetaAttribute on table ServiceMetaAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationBinder bindClass
> INFO: Binding entity from annotated class: jp.go.nict.langrid.dao.entity.ServiceInterfaceDefinition
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.EntityBinder bindTable
> INFO: Bind entity jp.go.nict.langrid.dao.entity.ServiceInterfaceDefinition on table ServiceInterfaceDefinition
> Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
> INFO: Using dialect: org.hibernate.dialect.HSQLDialect
> Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
> INFO: Using dialect: org.hibernate.dialect.HSQLDialect
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Grid.attributes -> GridAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.User.attributes -> UserAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.User.roles -> userRoles
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Resource.attributes -> ResourceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.attributes -> ServiceAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.deployments -> ServiceDeployment
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.endpoints -> ServiceEndpoint
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Service.invocations -> Invocation
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.Node.attributes -> NodeAttribute
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.annotations.CollectionBinder bindOneToManySecondPass
> INFO: Mapping collection: jp.go.nict.langrid.dao.entity.ServiceType.interfaceDefinitions -> ServiceInterfaceDefinition
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.AnnotationConfiguration secondPassCompile
> INFO: Hibernate Validator not found: ignoring
> Feb 18, 2017 11:03:12 AM org.hibernate.util.NamingHelper getInitialContext
> INFO: JNDI InitialContext properties:{}
> Feb 18, 2017 11:03:12 AM org.hibernate.connection.DatasourceConnectionProvider configure
> INFO: Using datasource: java:comp/env/jdbc/langrid
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> WARNING: Could not obtain connection to query metadata
> org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
> 	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
> 	at org.hibernate.cfg.SettingsFactory.buildSettings(SettingsFactory.java:111)
> 	at org.hibernate.cfg.Configuration.buildSettingsInternal(Configuration.java:2119)
> 	at org.hibernate.cfg.Configuration.buildSettings(Configuration.java:2115)
> 	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1339)
> 	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
> 	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
> 	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
> 	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
> 	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
> 	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
> 	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
> 	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
> 	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
> 	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
> 	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
> 	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
> 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
> 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
> 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
> 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
> 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
> 	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
> 	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
> 	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
> 	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
> 	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
> 	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
> 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
> Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
> 	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
> 	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
> 	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
> 	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
> 	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
> 	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
> 	at org.postgresql.Driver.makeConnection(Driver.java:391)
> 	at org.postgresql.Driver.connect(Driver.java:265)
> 	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
> 	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
> 	... 50 more
> 
> Feb 18, 2017 11:03:12 AM org.hibernate.dialect.Dialect <init>
> INFO: Using dialect: org.hibernate.dialect.PostgreSQLDialect
> Feb 18, 2017 11:03:12 AM org.hibernate.transaction.TransactionFactoryFactory buildTransactionFactory
> INFO: Using default transaction strategy (direct JDBC transactions)
> Feb 18, 2017 11:03:12 AM org.hibernate.transaction.TransactionManagerLookupFactory getTransactionManagerLookup
> INFO: No TransactionManagerLookup configured (in JTA environment, use of read-write or transactional second-level cache is not recommended)
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Automatic flush during beforeCompletion(): disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Automatic session close at end of transaction: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Scrollable result sets: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: JDBC3 getGeneratedKeys(): disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Connection release mode: auto
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Default batch fetch size: 1
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Generate SQL with comments: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Order SQL updates by primary key: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Order SQL inserts for batching: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory createQueryTranslatorFactory
> INFO: Query translator: org.hibernate.hql.ast.ASTQueryTranslatorFactory
> Feb 18, 2017 11:03:12 AM org.hibernate.hql.ast.ASTQueryTranslatorFactory <init>
> INFO: Using ASTQueryTranslatorFactory
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Query language substitutions: {}
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: JPA-QL strict compliance: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Second-level cache: enabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Query cache: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory createRegionFactory
> INFO: Cache region factory : org.hibernate.cache.impl.bridge.RegionFactoryCacheProviderBridge
> Feb 18, 2017 11:03:12 AM org.hibernate.cache.impl.bridge.RegionFactoryCacheProviderBridge <init>
> INFO: Cache provider: org.hibernate.cache.NoCacheProvider
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Optimize cache for minimal puts: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Structured second-level cache entries: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Statistics: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Deleted entity synthetic identifier rollback: disabled
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Default entity-mode: pojo
> Feb 18, 2017 11:03:12 AM org.hibernate.cfg.SettingsFactory buildSettings
> INFO: Named query checking : enabled
> Feb 18, 2017 11:03:12 AM org.hibernate.impl.SessionFactoryImpl <init>
> INFO: building session factory
> Feb 18, 2017 11:03:12 AM org.hibernate.impl.SessionFactoryObjectFactory addInstance
> INFO: Not binding factory to JNDI, no JNDI name configured
> Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
> INFO: Running hbm2ddl schema update
> Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
> INFO: fetching database metadata
> Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
> SEVERE: could not get database metadata
> org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
> 	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
> 	at org.hibernate.tool.hbm2ddl.SuppliedConnectionProviderConnectionHelper.prepare(SuppliedConnectionProviderConnectionHelper.java:51)
> 	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:168)
> 	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:356)
> 	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1341)
> 	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
> 	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
> 	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
> 	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
> 	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
> 	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
> 	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
> 	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
> 	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
> 	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
> 	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
> 	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
> 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
> 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
> 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
> 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
> 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
> 	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
> 	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
> 	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
> 	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
> 	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
> 	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
> 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
> Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
> 	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
> 	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
> 	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
> 	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
> 	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
> 	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
> 	at org.postgresql.Driver.makeConnection(Driver.java:391)
> 	at org.postgresql.Driver.connect(Driver.java:265)
> 	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
> 	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
> 	... 50 more
> 
> Feb 18, 2017 11:03:12 AM org.hibernate.tool.hbm2ddl.SchemaUpdate execute
> SEVERE: could not complete schema update
> org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
> 	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
> 	at org.hibernate.tool.hbm2ddl.SuppliedConnectionProviderConnectionHelper.prepare(SuppliedConnectionProviderConnectionHelper.java:51)
> 	at org.hibernate.tool.hbm2ddl.SchemaUpdate.execute(SchemaUpdate.java:168)
> 	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:356)
> 	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1341)
> 	at org.hibernate.cfg.AnnotationConfiguration.buildSessionFactory(AnnotationConfiguration.java:867)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.initialize(HibernateDaoContext.java:571)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.getSession(HibernateDaoContext.java:301)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:142)
> 	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
> 	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
> 	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
> 	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
> 	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
> 	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
> 	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
> 	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
> 	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
> 	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
> 	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
> 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
> 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
> 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
> 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
> 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
> 	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
> 	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
> 	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
> 	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
> 	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
> 	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
> 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
> Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
> 	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
> 	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
> 	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
> 	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
> 	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
> 	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
> 	at org.postgresql.Driver.makeConnection(Driver.java:391)
> 	at org.postgresql.Driver.connect(Driver.java:265)
> 	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
> 	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
> 	... 50 more
> 
> Feb 18, 2017 11:03:13 AM org.hibernate.dialect.Dialect <init>
> INFO: Using dialect: org.hibernate.dialect.PostgreSQLDialect
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> WARNING: SQL Error: 0, SQLState: null
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> 11:03:13,119 ERROR ServiceManagerApplication:124 - [TreadID:1,UserID:Spring Transaction,ClassName:ExtendTransactionInterceptor]
> jp.go.nict.langrid.dao.ConnectException: org.hibernate.exception.GenericJDBCException: Cannot open connection
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:151)
> 	at jp.go.nict.langrid.management.web.spring.ExtendTransactionInterceptor.invoke(ExtendTransactionInterceptor.java:23)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
> 	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
> 	at com.sun.proxy.$Proxy9.setSelfGridId(Unknown Source)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.wicket.proxy.LazyInitProxyFactory$JdkHandler.invoke(LazyInitProxyFactory.java:414)
> 	at org.apache.wicket.proxy.$Proxy33.setSelfGridId(Unknown Source)
> 	at jp.go.nict.langrid.management.web.view.ServiceManagerApplication.init(ServiceManagerApplication.java:216)
> 	at org.apache.wicket.protocol.http.WicketFilter.init(WicketFilter.java:726)
> 	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:295)
> 	at org.apache.catalina.core.ApplicationFilterConfig.setFilterDef(ApplicationFilterConfig.java:422)
> 	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:115)
> 	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4072)
> 	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4726)
> 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:799)
> 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:779)
> 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:675)
> 	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:601)
> 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:502)
> 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
> 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
> 	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
> 	at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
> 	at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
> 	at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
> 	at org.apache.catalina.core.StandardService.start(StandardService.java:525)
> 	at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
> 	at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
> 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
> Caused by: org.hibernate.exception.GenericJDBCException: Cannot open connection
> 	at org.hibernate.exception.SQLStateConverter.handledNonSpecificException(SQLStateConverter.java:126)
> 	at org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:114)
> 	at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:66)
> 	at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:52)
> 	at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:449)
> 	at org.hibernate.jdbc.ConnectionManager.getConnection(ConnectionManager.java:167)
> 	at org.hibernate.jdbc.JDBCContext.connection(JDBCContext.java:142)
> 	at org.hibernate.transaction.JDBCTransaction.begin(JDBCTransaction.java:85)
> 	at org.hibernate.impl.SessionImpl.beginTransaction(SessionImpl.java:1354)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.hibernate.context.ThreadLocalSessionContext$TransactionProtectionWrapper.invoke(ThreadLocalSessionContext.java:342)
> 	at com.sun.proxy.$Proxy73.beginTransaction(Unknown Source)
> 	at jp.go.nict.langrid.dao.hibernate.HibernateDaoContext.beginTransaction(HibernateDaoContext.java:147)
> 	... 39 more
> Caused by: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
> 	at org.hibernate.connection.DatasourceConnectionProvider.getConnection(DatasourceConnectionProvider.java:92)
> 	at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:446)
> 	... 50 more
> Caused by: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user "langrid"
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:291)
> 	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:106)
> 	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:64)
> 	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:123)
> 	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:28)
> 	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:20)
> 	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
> 	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:22)
> 	at org.postgresql.Driver.makeConnection(Driver.java:391)
> 	at org.postgresql.Driver.connect(Driver.java:265)
> 	at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
> 	at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
> 	at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
> 	... 54 more
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> WARNING: SQL Error: 0, SQLState: null
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> WARNING: SQL Error: 0, SQLState: null
> Feb 18, 2017 11:03:13 AM org.hibernate.util.JDBCExceptionReporter logExceptions
> SEVERE: Cannot create PoolableConnectionFactory (FATAL: Ident authentication failed for user "langrid")
> Feb 18, 2017 11:03:13 AM org.apache.catalina.core.StandardContext start
> SEVERE: Error filterStart
> Feb 18, 2017 11:03:13 AM org.apache.catalina.core.StandardContext start
> SEVERE: Context [/service_manager] startup failed due to previous errors
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-1] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-2] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-3] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-4] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-5] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-6] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-7] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-8] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-9] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [org.springframework.scheduling.quartz.SchedulerFactoryBean#0_Worker-10] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [Timer-0] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearReferencesThreads
> SEVERE: The web application [/service_manager] appears to have started a thread named [PageSavingThread-ServiceManagerFilter] but has failed to stop it. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.catalina.loader.WebappClassLoader clearThreadLocalMap
> SEVERE: The web application [/service_manager] created a ThreadLocal with key of type [jp.go.nict.langrid.dao.hibernate.HibernateDaoContext$1] (value [jp.go.nict.langrid.dao.hibernate.HibernateDaoContext$1@779de014]) and a value of type [java.lang.Integer] (value [0]) but failed to remove it when the web application was stopped. This is very likely to create a memory leak.
> Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
> INFO: registered listener interface [RequestListenerInterface name=ILinkListener, method=public abstract void org.apache.wicket.markup.html.link.ILinkListener.onLinkClicked()]
> Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
> INFO: registered listener interface [RequestListenerInterface name=INewBrowserWindowListener, method=public abstract void org.apache.wicket.markup.html.INewBrowserWindowListener.onNewBrowserWindow()]
> Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
> INFO: registered listener interface [RequestListenerInterface name=IResourceListener, method=public abstract void org.apache.wicket.IResourceListener.onResourceRequested()]
> Feb 18, 2017 11:03:13 AM org.apache.wicket.RequestListenerInterface registerRequestListenerInterface
> INFO: registered listener interface [RequestListenerInterface name=IRedirectListener, method=public abstract void org.apache.wicket.IRedirectListener.onRedirect()]
> Feb 18, 2017 11:03:13 AM org.apache.catalina.startup.HostConfig deployDirectory
> INFO: Deploying web application directory ROOT
> Feb 18, 2017 11:03:13 AM org.apache.coyote.http11.Http11Protocol start
> INFO: Starting Coyote HTTP/1.1 on http-8080
> Feb 18, 2017 11:03:13 AM org.apache.jk.common.ChannelSocket init
> INFO: JK: ajp13 listening on /0.0.0.0:8009
> Feb 18, 2017 11:03:13 AM org.apache.jk.server.JkMain start
> INFO: Jk running ID=0 time=0/8  config=null
> Feb 18, 2017 11:03:13 AM org.apache.catalina.startup.Catalina start
> INFO: Server startup in 2772 ms
> Feb 18, 2017 11:03:24 AM org.apache.catalina.startup.HostConfig checkResources
> INFO: Undeploying context [/host-manager]
> Feb 18, 2017 11:03:24 AM org.apache.catalina.startup.HostConfig checkResources
> INFO: Undeploying context [/manager]
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/lappsgrid-incubator/service-manager-installation/issues/17>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AD6IGLQE6vQmRJb2UKtSARHcMb2vec0gks5rdxgpgaJpZM4MFKf1>.
>

--- a/after.sh
+++ b/after.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-psql $DATABASE < /tmp/create_indices.sql
-psql $DATABASE < /tmp/create_storedproc.sql
+psql -U postgres $DATABASE < /tmp/create_indices.sql
+psql -U postgres $DATABASE < /tmp/create_storedproc.sql
 echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
 #psql $DATABSE < /tmp/alter.sql
 cat /tmp/alter.sql | psql -U postgres $DATABASE

--- a/after.sh
+++ b/after.sh
@@ -1,17 +1,80 @@
 #!/usr/bin/env bash
+set -u
 
-log "Creating indices and stored procedure."
-# Grab the SQL and setup scripts.
-wget $manager/create_storedproc.sql
+smg=smg-1.1.0-SNAPSHOT
+manager=http://downloads.lappsgrid.org/service-manager
+scripts=http://downloads.lappsgrid.org/scripts
+
+# Locations that Tomcat is installed.
+MANAGER=/usr/share/tomcat/service-manager
+BPEL=/usr/share/tomcat/active-bpel
+
+function usage()
+{
+	echo
+	echo "USAGE"
+	echo "   sudo ./setup.sh"
+	echo
+	echo
+}
+
+function log {
+	echo $1
+	echo "$(date +'%b %d %Y %Y') - $1" >> /var/log/service-manager-install.log 
+}
+
+function start_tomcat {
+	log "Staring tomcat."
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
+		systemctl start tomcat
+	else
+		service tomcat start
+	fi
+}
+
+function stop_tomcat {
+	log "Stopping tomcat."
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
+		systemctl stop tomcat
+	else
+		service tomcat stop
+	fi
+}
+
+function wait_for {
+	while ! grep "Server startup in" $1/logs/catalina.out ; do
+		log "Waiting for $1 to start"
+		sleep 3
+	done
+}
+
+function toggle_tomcat {
+	start_tomcat
+	wait_for $MANAGER
+	wait_for $BPEL
+	stop_tomcat
+}
+
+source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
+
+if [[ -z $OS ]] ; then
+	log "The variable OS has not been set!"
+	exit 1
+fi
+
+log "Creating indices."
 wget $manager/create_indices.sql
+cat create_indices.sql | sudo -u postgres psql $DATABASE
 
-psql -U postgres $DATABASE < /tmp/create_indices.sql
-psql -U postgres $DATABASE < /tmp/create_storedproc.sql
-echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
-#psql $DATABSE < /tmp/alter.sql
-cat /tmp/alter.sql | psql -U postgres $DATABASE
+log "Creating stored procedure."
+wget $manager/create_storedproc.sql
+cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
 
-#psql $DATABASE -c "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME"
+# We need to generate this on the fly since it include the user
+# defined ROLENAME.
+log "Changing owner of the stored procedure."
+echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > alter.sql
+cat alter.sql | sudo -u postgres psql $DATABASE
 
 log "Securing the Tomcat installations"
 for dir in $MANAGER $BPEL ; do
@@ -29,22 +92,19 @@ for dir in $MANAGER $BPEL ; do
 	popd > /dev/null
 done
 
-log "Downloading the latest service manager war file."
-wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
-mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
+#log "Removing default webapps."
+#for dir in $MANAGER/webapps $BPEL/webapps ; do
+#	pushd $dir > /dev/null
+#	rm -rf docs examples manager host-manager
+#	popd > /dev/null
+#done
 
-log "Removing default webapps."
-for dir in $MANAGER/webapps $BPEL/webapps ; do
-	pushd $dir > /dev/null
-	rm -rf docs examples manager host-manager
-	popd > /dev/null
-done
-
-if [[ $OS = redhat7 ]] ; then
-	systemctl start tomcat
-else
-	service tomcat start
+if [[ $OS = redhat7 || $OS = centos ]] ; then
+	log "Opening port 8080"
+	iptables -I INPUT -p tcp -m tcp --dport 8080 -j ACCEPT
 fi
+
+start_tomcat 
 
 log "The Service Grid is now running."
 echo

--- a/after.sh
+++ b/after.sh
@@ -1,0 +1,45 @@
+#!/usr/bin/env bash
+
+psql $DATABASE < /tmp/create_indices.sql
+psql $DATABASE < /tmp/create_storedproc.sql
+echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
+#psql $DATABSE < /tmp/alter.sql
+cat /tmp/alter.sql | psql -U postgres $DATABASE
+
+#psql $DATABASE -c "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME"
+
+log "Securing the Tomcat installations"
+for dir in $MANAGER $BPEL ; do
+	pushd $dir > /dev/null
+	# Make the tomcat user the owner of everything.
+	chown -R tomcat:tomcat .
+
+	# Tighten permissions on subdirectories.
+	chmod -R 0700 bin
+	chmod -R 0700 conf
+	chmod -R 0750 logs
+	chmod -R 0700 temp
+	chmod -R 0700 work
+	chmod -R 0770 webapps
+	popd > /dev/null
+done
+
+log "Downloading the latest service manager war file."
+wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
+mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
+
+log "Removing default webapps."
+for dir in $MANAGER/webapps $BPEL/webapps ; do
+	pushd $dir > /dev/null
+	rm -rf docs examples manager host-manager
+	popd > /dev/null
+done
+
+if [[ $OS = redhat7 ]] ; then
+	systemctl start tomcat
+else
+	service tomcat start
+fi
+
+log "The Service Grid is now running."
+echo

--- a/after.sh
+++ b/after.sh
@@ -1,13 +1,14 @@
 #!/usr/bin/env bash
 set -u
 
-smg=smg-1.1.0-SNAPSHOT
-manager=http://downloads.lappsgrid.org/service-manager
-scripts=http://downloads.lappsgrid.org/scripts
+export SMG=smg-1.1.0-SNAPSHOT
+#manager=http://downloads.lappsgrid.org/service-manager
+export MANAGER=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+export SCRIPTS=http://downloads.lappsgrid.org/scripts
 
 # Locations that Tomcat is installed.
-MANAGER=/usr/share/tomcat/service-manager
-BPEL=/usr/share/tomcat/active-bpel
+TOMCAT_MANAGER=/usr/share/tomcat/service-manager
+TOMCAT_BPEL=/usr/share/tomcat/active-bpel
 
 function usage()
 {
@@ -50,9 +51,11 @@ function wait_for {
 
 function toggle_tomcat {
 	start_tomcat
-	wait_for $MANAGER
-	wait_for $BPEL
+	sleep 1
+	wait_for $TOMCAT_MANAGER
+	wait_for $TOMCAT_BPEL
 	stop_tomcat
+	sleep 5
 }
 
 source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)

--- a/after.sh
+++ b/after.sh
@@ -78,6 +78,7 @@ wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/release
 mv `ls *.war | head -1` $TOMCAT_MANAGER/webapps/service_manager.war
 
 toggle_tomcat
+source db.confg
 
 log "Creating indices."
 wget $MANAGER/create_indices.sql

--- a/after.sh
+++ b/after.sh
@@ -62,12 +62,26 @@ if [[ -z $OS ]] ; then
 	exit 1
 fi
 
+cp tomcat-users.xml $TOMCAT_MANAGER/conf
+cp service_manager.xml $TOMCAT_MANAGER/conf/Catalina/localhost
+
+cp tomcat-users-bpel.xml $TOMCAT_BPEL/conf/tomcat-users.xml
+cp active-bpel.xml $TOMCAT_BPEL/conf/Catalina/localhost
+cp langrid.ae.properties $TOMCAT_BPEL/bpr
+
+# Get the new .war file before starting Tomcat for the first time.
+log "Downloading the latest service manager war file."
+wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
+mv `ls *.war | head -1` $TOMCAT_MANAGER/webapps/service_manager.war
+
+toggle_tomcat
+
 log "Creating indices."
-wget $manager/create_indices.sql
+wget $MANAGER/create_indices.sql
 cat create_indices.sql | sudo -u postgres psql $DATABASE
 
 log "Creating stored procedure."
-wget $manager/create_storedproc.sql
+wget $MANAGER/create_storedproc.sql
 cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
 
 # We need to generate this on the fly since it include the user
@@ -77,7 +91,7 @@ echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varyi
 cat alter.sql | sudo -u postgres psql $DATABASE
 
 log "Securing the Tomcat installations"
-for dir in $MANAGER $BPEL ; do
+for dir in $TOMCAT_MANAGER $TOMCAT_BPEL ; do
 	pushd $dir > /dev/null
 	# Make the tomcat user the owner of everything.
 	chown -R tomcat:tomcat .
@@ -93,7 +107,7 @@ for dir in $MANAGER $BPEL ; do
 done
 
 #log "Removing default webapps."
-#for dir in $MANAGER/webapps $BPEL/webapps ; do
+#for dir in $TOMCAT_MANAGER/webapps $TOMCAT_BPEL/webapps ; do
 #	pushd $dir > /dev/null
 #	rm -rf docs examples manager host-manager
 #	popd > /dev/null

--- a/after.sh
+++ b/after.sh
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+log "Creating indices and stored procedure."
+# Grab the SQL and setup scripts.
+wget $manager/create_storedproc.sql
+wget $manager/create_indices.sql
+
 psql -U postgres $DATABASE < /tmp/create_indices.sql
 psql -U postgres $DATABASE < /tmp/create_storedproc.sql
 echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql

--- a/before.sh
+++ b/before.sh
@@ -78,6 +78,7 @@ chmod +x $smg/smg
 # 	langrid.ae.properties
 # 	db.config
 $smg/smg ServiceManager.config
+source db.config
 
 #wget $manager/database-setup.sh
 

--- a/before.sh
+++ b/before.sh
@@ -1,14 +1,14 @@
 #!/usr/bin/env bash
 set -u
 
-smg=smg-1.1.0-SNAPSHOT
+export SMG=smg-1.1.0-SNAPSHOT
 #manager=http://downloads.lappsgrid.org/service-manager
-manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
-scripts=http://downloads.lappsgrid.org/scripts
+export MANAGER=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+export SCRIPTS=http://downloads.lappsgrid.org/scripts
 
 # Locations that Tomcat is installed.
-MANAGER=/usr/share/tomcat/service-manager
-BPEL=/usr/share/tomcat/active-bpel
+TOMCAT_MANAGER=/usr/share/tomcat/service-manager
+TOMCAT_BPEL=/usr/share/tomcat/active-bpel
 
 function usage()
 {
@@ -51,9 +51,10 @@ function wait_for {
 
 function toggle_tomcat {
 	start_tomcat
-	wait_for $MANAGER
-	wait_for $BPEL
+	wait_for $TOMCAT_MANAGER
+	wait_for $TOMCAT_BPEL
 	stop_tomcat
+	sleep 5
 }
 
 source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
@@ -81,20 +82,20 @@ set -eu
 
 # Installs the packages required to install and run the Service Grid.
 log "Installing common packages"
-curl -sSL $scripts/install-common.sh | bash
+curl -sSL $SCRIPTS/install-common.sh | bash
 
 # Edit the properties file to get it out of the way and allow then
 # rest of the script to continue uninterrupted.
 log "Configuring the Service Manager"
 if [[ ! -e service-manager.properties ]] ; then
-	wget $manager/service-manager.properties
+	wget $MANAGER/service-manager.properties
 fi
 $EDITOR service-manager.properties
 
 log "Installing Java"
-curl -sSL $scripts/install-java.sh | bash
+curl -sSL $SCRIPTS/install-java.sh | bash
 log "Installing PostgreSQL"
-curl -sSL $scripts/install-postgres.sh | bash
+curl -sSL $SCRIPTS/install-postgres.sh | bash
 
 if [[ $OS = centos || $OS = redhat7 ]] ; then
 	hba=/var/lib/pgsql/9.6/data/pg_hba.conf
@@ -106,14 +107,14 @@ if [[ $OS = centos || $OS = redhat7 ]] ; then
 fi
 
 if [ ! -e ServiceManager.config ] ; then
-	wget $manager/ServiceManager.config
+	wget $MANAGER/ServiceManager.config
 fi
 
 # Get the program used to transform the ServiceManager.config file
 # into the various xml files.
-wget http://downloads.lappsgrid.org/$smg.tgz
-tar xzf $smg.tgz
-chmod +x $smg/smg
+wget http://downloads.lappsgrid.org/$SMG.tgz
+tar xzf $SMG.tgz
+chmod +x $SMG/smg
 
 # Processing the ServiceManager.config will generate:
 # 	service_manager.xml
@@ -122,7 +123,7 @@ chmod +x $smg/smg
 # 	tomcat-users-bpel.xml
 # 	langrid.ae.properties
 # 	db.config
-$smg/smg ServiceManager.config
+$SMG/smg ServiceManager.config
 source db.config
 
 sudo -u postgres createuser -S -D -R $ROLENAME
@@ -131,18 +132,5 @@ sudo -u postgres createdb $DATABASE -O $ROLENAME -E 'UTF8'
 
 # Now install Tomcat and create the PostgreSQL database.
 log "Starting Tomcat installation."
-curl -sSL $manager/install-tomcat.sh | bash
+curl -sSL $MANAGER/install-tomcat.sh | bash
 
-cp tomcat-users.xml $MANAGER/conf
-cp service_manager.xml $MANAGER/conf/Catalina/localhost
-
-cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
-cp active-bpel.xml $BPEL/conf/Catalina/localhost
-cp langrid.ae.properties $BPEL/bpr
-
-# Get the new .war file before starting Tomcat for the first time.
-log "Downloading the latest service manager war file."
-wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
-mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
-
-toggle_tomcat

--- a/before.sh
+++ b/before.sh
@@ -2,7 +2,8 @@
 set -u
 
 smg=smg-1.1.0-SNAPSHOT
-manager=http://downloads.lappsgrid.org/service-manager
+#manager=http://downloads.lappsgrid.org/service-manager
+manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
 scripts=http://downloads.lappsgrid.org/scripts
 
 # Locations that Tomcat is installed.

--- a/before.sh
+++ b/before.sh
@@ -1,0 +1,108 @@
+#!/usr/bin/env bash
+set -u
+
+smg=smg-1.1.0-SNAPSHOT
+manager=http://downloads.lappsgrid.org/service-manager
+scripts=http://downloads.lappsgrid.org/scripts
+
+function usage()
+{
+	echo
+	echo "USAGE"
+	echo "   sudo ./setup.sh"
+	echo
+	echo
+}
+
+function log {
+	echo $1 | tee /var/log/service-manager-install.log >&2
+}
+
+source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
+
+if [[ -z $OS ]] ; then
+	log "The variable OS has not been set!"
+	exit 1
+fi
+
+if [[ $OS = ubuntu ]] ; then
+	log "Updating apt-get indices."
+	apt-get update && apt-get upgrade -y
+elif [[ $OS = redhat || $OS = centos ]] ; then
+	yum makecache fast
+else
+	log "Unsupport Linux flavor: $OS"
+	exit 1
+fi
+
+set +u
+if [[ -z "$EDITOR" ]] ; then
+	EDITOR=emacs
+fi
+set -eu
+
+# Installs the packages required to install and run the Service Grid.
+log "Installing common packages"
+curl -sSL $scripts/install-common.sh | bash
+
+# Edit the properties file to get it out of the way and allow then
+# rest of the script to continue uninterrupted.
+log "Configuring the Service Manager"
+if [[ ! -e service-manager.properties ]] ; then
+	wget $manager/service-manager.properties
+fi
+$EDITOR service-manager.properties
+
+log "Installing Java"
+curl -sSL $scripts/install-java.sh | bash
+log "Installing PostgreSQL"
+curl -sSL $scripts/install-postgres.sh | bash
+
+if [ ! -e ServiceManager.config ] ; then
+	wget $manager/ServiceManager.config
+fi
+
+# Get the program used to transform the ServiceManager.config file
+# into the various xml files.
+wget http://downloads.lappsgrid.org/$smg.tgz
+tar xzf $smg.tgz
+chmod +x $smg/smg
+
+# Processing the ServiceManager.config will generate:
+# 	service_manager.xml
+# 	active-bpel.xml
+# 	tomcat-users.xml
+# 	tomcat-users-bpel.xml
+# 	langrid.ae.properties
+# 	db.config
+$smg/smg ServiceManager.config
+
+# Now install Tomcat and create the PostgreSQL database.
+log "Starting Tomcat installation."
+MANAGER=/usr/share/tomcat/service-manager
+BPEL=/usr/share/tomcat/active-bpel
+curl -sSL $manager/install-tomcat.sh | bash
+
+cp tomcat-users.xml $MANAGER/conf
+cp service_manager.xml $MANAGER/conf/Catalina/localhost
+
+cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
+cp active-bpel.xml $BPEL/conf/Catalina/localhost
+cp langrid.ae.properties $BPEL/bpr
+
+log "Creating role, database and stored procedure."
+# Grab the SQL and setup scripts.
+wget $manager/create_storedproc.sql
+wget $manager/create_indices.sql
+wget $manager/database-setup.sh
+
+# Generate a custom database-setup.sh with the settings generated above.
+#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
+#chmod +x /tmp/database-setup.sh
+#mv *.sql /tmp
+# Now run the setup script as the postgres user.
+#su - postgres -c "bash /tmp/database-setup.sh"
+#rm /tmp/*.sql
+createuser -U postgres -S -D -R $ROLENAME
+psql -U postgres --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
+createdb -U postgres $DATABASE -O $ROLENAME -E 'UTF8'

--- a/before.sh
+++ b/before.sh
@@ -5,6 +5,10 @@ smg=smg-1.1.0-SNAPSHOT
 manager=http://downloads.lappsgrid.org/service-manager
 scripts=http://downloads.lappsgrid.org/scripts
 
+# Locations that Tomcat is installed.
+MANAGER=/usr/share/tomcat/service-manager
+BPEL=/usr/share/tomcat/active-bpel
+
 function usage()
 {
 	echo
@@ -15,7 +19,40 @@ function usage()
 }
 
 function log {
-	echo $1 | tee /var/log/service-manager-install.log >&2
+	echo $1
+	echo "$(date +'%b %d %Y %Y') - $1" >> /var/log/service-manager-install.log 
+}
+
+function start_tomcat {
+	log "Staring tomcat."
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
+		systemctl start tomcat
+	else
+		service tomcat start
+	fi
+}
+
+function stop_tomcat {
+	log "Stopping tomcat."
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
+		systemctl stop tomcat
+	else
+		service tomcat stop
+	fi
+}
+
+function wait_for {
+	while ! grep "Server startup in" $1/logs/catalina.out ; do
+		log "Waiting for $1 to start"
+		sleep 3
+	done
+}
+
+function toggle_tomcat {
+	start_tomcat
+	wait_for $MANAGER
+	wait_for $BPEL
+	stop_tomcat
 }
 
 source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
@@ -58,7 +95,14 @@ curl -sSL $scripts/install-java.sh | bash
 log "Installing PostgreSQL"
 curl -sSL $scripts/install-postgres.sh | bash
 
-# Create the database and role needed by the service manager.
+if [[ $OS = centos || $OS = redhat7 ]] ; then
+	hba=/var/lib/pgsql/9.6/data/pg_hba.conf
+	rm $hba
+	echo "local all all trust" > $hba
+	echo "host all all 127.0.0.1/32 trust" >> $hba
+	echo "host all all ::1/128 trust" >> $hba
+	systemctl restart postgresql-9.6
+fi
 
 if [ ! -e ServiceManager.config ] ; then
 	wget $manager/ServiceManager.config
@@ -80,23 +124,12 @@ chmod +x $smg/smg
 $smg/smg ServiceManager.config
 source db.config
 
-#wget $manager/database-setup.sh
-
-# Generate a custom database-setup.sh with the settings generated above.
-#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
-#chmod +x /tmp/database-setup.sh
-#mv *.sql /tmp
-# Now run the setup script as the postgres user.
-#su - postgres -c "bash /tmp/database-setup.sh"
-#rm /tmp/*.sql
-createuser -U postgres -S -D -R $ROLENAME
-psql -U postgres --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
-createdb -U postgres $DATABASE -O $ROLENAME -E 'UTF8'
+sudo -u postgres createuser -S -D -R $ROLENAME
+sudo -u postgres psql --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
+sudo -u postgres createdb $DATABASE -O $ROLENAME -E 'UTF8'
 
 # Now install Tomcat and create the PostgreSQL database.
 log "Starting Tomcat installation."
-MANAGER=/usr/share/tomcat/service-manager
-BPEL=/usr/share/tomcat/active-bpel
 curl -sSL $manager/install-tomcat.sh | bash
 
 cp tomcat-users.xml $MANAGER/conf
@@ -106,3 +139,9 @@ cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
 cp active-bpel.xml $BPEL/conf/Catalina/localhost
 cp langrid.ae.properties $BPEL/bpr
 
+# Get the new .war file before starting Tomcat for the first time.
+log "Downloading the latest service manager war file."
+wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
+mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
+
+toggle_tomcat

--- a/before.sh
+++ b/before.sh
@@ -58,6 +58,8 @@ curl -sSL $scripts/install-java.sh | bash
 log "Installing PostgreSQL"
 curl -sSL $scripts/install-postgres.sh | bash
 
+# Create the database and role needed by the service manager.
+
 if [ ! -e ServiceManager.config ] ; then
 	wget $manager/ServiceManager.config
 fi
@@ -77,6 +79,19 @@ chmod +x $smg/smg
 # 	db.config
 $smg/smg ServiceManager.config
 
+#wget $manager/database-setup.sh
+
+# Generate a custom database-setup.sh with the settings generated above.
+#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
+#chmod +x /tmp/database-setup.sh
+#mv *.sql /tmp
+# Now run the setup script as the postgres user.
+#su - postgres -c "bash /tmp/database-setup.sh"
+#rm /tmp/*.sql
+createuser -U postgres -S -D -R $ROLENAME
+psql -U postgres --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
+createdb -U postgres $DATABASE -O $ROLENAME -E 'UTF8'
+
 # Now install Tomcat and create the PostgreSQL database.
 log "Starting Tomcat installation."
 MANAGER=/usr/share/tomcat/service-manager
@@ -90,19 +105,3 @@ cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
 cp active-bpel.xml $BPEL/conf/Catalina/localhost
 cp langrid.ae.properties $BPEL/bpr
 
-log "Creating role, database and stored procedure."
-# Grab the SQL and setup scripts.
-wget $manager/create_storedproc.sql
-wget $manager/create_indices.sql
-wget $manager/database-setup.sh
-
-# Generate a custom database-setup.sh with the settings generated above.
-#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
-#chmod +x /tmp/database-setup.sh
-#mv *.sql /tmp
-# Now run the setup script as the postgres user.
-#su - postgres -c "bash /tmp/database-setup.sh"
-#rm /tmp/*.sql
-createuser -U postgres -S -D -R $ROLENAME
-psql -U postgres --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
-createdb -U postgres $DATABASE -O $ROLENAME -E 'UTF8'

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -39,3 +39,5 @@ else
 fi
 
 echo "Tomcat installed to /usr/share/tomcat"
+service tomcat status
+

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -43,5 +43,4 @@ else
 fi
 
 echo "Tomcat installed to /usr/share/tomcat"
-service tomcat status
 

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -2,9 +2,9 @@
 
 # The manager variable should be picked up from the calling environment.  If not
 # we set it to a likely default value.
-if [[ -z $manager ]] ; then
+if [[ -z $MANAGER ]] ; then
 	#manager=http://downloads.lappsgrid.org/service-manager
-	manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+	MANAGER=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
 fi
 
 if [ -z "$OS" ] ; then
@@ -28,12 +28,12 @@ chown -R tomcat:tomcat /usr/share/tomcat
 
 
 if [[ $OS = centos ]] ; then
-    wget $manager/tomcat.service
+    wget $MANAGER/tomcat.service
     mv tomcat.service /etc/systemd/system/
     systemctl daemon-reload
     systemctl enable tomcat.service
 elif [[ $OS = ubuntu || $OS = redhat ]] ; then
-    wget $manager/tomcat.sh
+    wget $MANAGER/tomcat.sh
     mv tomcat.sh /etc/init.d/tomcat
     chmod +x /etc/init.d/tomcat
 	update-rc.d tomcat defaults

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -11,7 +11,8 @@ if [ -z "$OS" ] ; then
 	source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
 fi
 
-if [[ ! `grep -c '^tomcat:' /etc/passwd` ]] ; then 
+#if [[ ! `grep -c '^tomcat:' /etc/passwd` ]] ; then 
+if ! id tomcat ; then
     if [[ $OS = redhat* || $OS = centos ]] ; then
         adduser -r -d /usr/share/tomcat -s /usr/bin/bash tomcat
     elif [[ $OS = ubuntu ]] ; then

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -1,7 +1,11 @@
 #!/usr/bin/env bash
 
-#manager=http://downloads.lappsgrid.org/service-manager
-manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+# The manager variable should be picked up from the calling environment.  If not
+# we set it to a likely default value.
+if [[ -z $manager ]] ; then
+	#manager=http://downloads.lappsgrid.org/service-manager
+	manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+fi
 
 if [ -z "$OS" ] ; then
 	source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -11,15 +11,17 @@ if [ -z "$OS" ] ; then
 	source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
 fi
 
-if [[ $OS = redhat || $OS = centos ]] ; then
-    adduser -r -d /usr/share/tomcat -s /usr/bin/bash tomcat
-elif [[ $OS = ubuntu ]] ; then
-    groupadd tomcat
-    adduser --system --home /usr/share/tomcat --shell /usr/bin/bash --ingroup tomcat tomcat
-else
-	echo "Unknown Linux flavor"
-	exit 1
-fi
+if [[ ! `grep -c '^tomcat:' /etc/passwd` ]] ; then 
+    if [[ $OS = redhat* || $OS = centos ]] ; then
+        adduser -r -d /usr/share/tomcat -s /usr/bin/bash tomcat
+    elif [[ $OS = ubuntu ]] ; then
+        groupadd tomcat
+        adduser --system --home /usr/share/tomcat --shell /usr/bin/bash --ingroup tomcat tomcat
+    else
+        echo "Unknown Linux flavor"
+        exit 1
+    fi
+fi 
 
 wget http://downloads.lappsgrid.org/tomcat.tgz
 tar xzf tomcat.tgz
@@ -27,12 +29,12 @@ mv tomcat /usr/share
 chown -R tomcat:tomcat /usr/share/tomcat
 
 
-if [[ $OS = centos ]] ; then
+if [[ $OS = centos || $OS = redhat7 ]] ; then
     wget $MANAGER/tomcat.service
     mv tomcat.service /etc/systemd/system/
     systemctl daemon-reload
     systemctl enable tomcat.service
-elif [[ $OS = ubuntu || $OS = redhat ]] ; then
+elif [[ $OS = ubuntu || $OS = redhat6 ]] ; then
     wget $MANAGER/tomcat.sh
     mv tomcat.sh /etc/init.d/tomcat
     chmod +x /etc/init.d/tomcat

--- a/install-tomcat.sh
+++ b/install-tomcat.sh
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+#manager=http://downloads.lappsgrid.org/service-manager
+manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+
 if [ -z "$OS" ] ; then
 	source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
 fi
@@ -21,12 +24,12 @@ chown -R tomcat:tomcat /usr/share/tomcat
 
 
 if [[ $OS = centos ]] ; then
-    wget http://downloads.lappsgrid.org/service-manager/tomcat.service
+    wget $manager/tomcat.service
     mv tomcat.service /etc/systemd/system/
     systemctl daemon-reload
     systemctl enable tomcat.service
 elif [[ $OS = ubuntu || $OS = redhat ]] ; then
-    wget http://downloads.lappsgrid.org/service-manager/tomcat.sh
+    wget $manager/tomcat.sh
     mv tomcat.sh /etc/init.d/tomcat
     chmod +x /etc/init.d/tomcat
 	update-rc.d tomcat defaults

--- a/setup.sh
+++ b/setup.sh
@@ -1,14 +1,14 @@
 #!/usr/bin/env bash
 set -u
 
-smg=smg-1.1.0-SNAPSHOT
+export SMG=smg-1.1.0-SNAPSHOT
 #manager=http://downloads.lappsgrid.org/service-manager
-manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
-scripts=http://downloads.lappsgrid.org/scripts
+export MANAGER=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
+export SCRIPTS=http://downloads.lappsgrid.org/scripts
 
 # Locations that Tomcat is installed.
-MANAGER=/usr/share/tomcat/service-manager
-BPEL=/usr/share/tomcat/active-bpel
+TOMCAT_MANAGER=/usr/share/tomcat/service-manager
+TOMCAT_BPEL=/usr/share/tomcat/active-bpel
 
 function usage()
 {
@@ -51,8 +51,8 @@ function wait_for {
 
 function toggle_tomcat {
 	start_tomcat
-	wait_for $MANAGER
-	wait_for $BPEL
+	wait_for $TOMCAT_MANAGER
+	wait_for $TOMCAT_BPEL
 	stop_tomcat
 	sleep 5
 }
@@ -82,20 +82,20 @@ set -eu
 
 # Installs the packages required to install and run the Service Grid.
 log "Installing common packages"
-curl -sSL $scripts/install-common.sh | bash
+curl -sSL $SCRIPTS/install-common.sh | bash
 
 # Edit the properties file to get it out of the way and allow then
 # rest of the script to continue uninterrupted.
 log "Configuring the Service Manager"
 if [[ ! -e service-manager.properties ]] ; then
-	wget $manager/service-manager.properties
+	wget $MANAGER/service-manager.properties
 fi
 $EDITOR service-manager.properties
 
 log "Installing Java"
-curl -sSL $scripts/install-java.sh | bash
+curl -sSL $SCRIPTS/install-java.sh | bash
 log "Installing PostgreSQL"
-curl -sSL $scripts/install-postgres.sh | bash
+curl -sSL $SCRIPTS/install-postgres.sh | bash
 
 if [[ $OS = centos || $OS = redhat7 ]] ; then
 	hba=/var/lib/pgsql/9.6/data/pg_hba.conf
@@ -107,14 +107,14 @@ if [[ $OS = centos || $OS = redhat7 ]] ; then
 fi
 
 if [ ! -e ServiceManager.config ] ; then
-	wget $manager/ServiceManager.config
+	wget $MANAGER/ServiceManager.config
 fi
 
 # Get the program used to transform the ServiceManager.config file
 # into the various xml files.
-wget http://downloads.lappsgrid.org/$smg.tgz
-tar xzf $smg.tgz
-chmod +x $smg/smg
+wget http://downloads.lappsgrid.org/$SMG.tgz
+tar xzf $SMG.tgz
+chmod +x $SMG/smg
 
 # Processing the ServiceManager.config will generate:
 # 	service_manager.xml
@@ -123,7 +123,7 @@ chmod +x $smg/smg
 # 	tomcat-users-bpel.xml
 # 	langrid.ae.properties
 # 	db.config
-$smg/smg ServiceManager.config
+$SMG/smg ServiceManager.config
 source db.config
 
 sudo -u postgres createuser -S -D -R $ROLENAME
@@ -132,28 +132,28 @@ sudo -u postgres createdb $DATABASE -O $ROLENAME -E 'UTF8'
 
 # Now install Tomcat and create the PostgreSQL database.
 log "Starting Tomcat installation."
-curl -sSL $manager/install-tomcat.sh | bash
+curl -sSL $MANAGER/install-tomcat.sh | bash
 
-cp tomcat-users.xml $MANAGER/conf
-cp service_manager.xml $MANAGER/conf/Catalina/localhost
+cp tomcat-users.xml $TOMCAT_MANAGER/conf
+cp service_manager.xml $TOMCAT_MANAGER/conf/Catalina/localhost
 
-cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
-cp active-bpel.xml $BPEL/conf/Catalina/localhost
-cp langrid.ae.properties $BPEL/bpr
+cp tomcat-users-bpel.xml $TOMCAT_BPEL/conf/tomcat-users.xml
+cp active-bpel.xml $TOMCAT_BPEL/conf/Catalina/localhost
+cp langrid.ae.properties $TOMCAT_BPEL/bpr
 
 # Get the new .war file before starting Tomcat for the first time.
 log "Downloading the latest service manager war file."
 wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
-mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
+mv `ls *.war | head -1` $TOMCAT_MANAGER/webapps/service_manager.war
 
 toggle_tomcat
 
 log "Creating indices."
-wget $manager/create_indices.sql
+wget $MANAGER/create_indices.sql
 cat create_indices.sql | sudo -u postgres psql $DATABASE
 
 log "Creating stored procedure."
-wget $manager/create_storedproc.sql
+wget $MANAGER/create_storedproc.sql
 cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
 
 # We need to generate this on the fly since it include the user
@@ -163,7 +163,7 @@ echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varyi
 cat alter.sql | sudo -u postgres psql $DATABASE
 
 log "Securing the Tomcat installations"
-for dir in $MANAGER $BPEL ; do
+for dir in $TOMCAT_MANAGER $TOMCAT_BPEL ; do
 	pushd $dir > /dev/null
 	# Make the tomcat user the owner of everything.
 	chown -R tomcat:tomcat .
@@ -179,7 +179,7 @@ for dir in $MANAGER $BPEL ; do
 done
 
 #log "Removing default webapps."
-#for dir in $MANAGER/webapps $BPEL/webapps ; do
+#for dir in $TOMCAT_MANAGER/webapps $TOMCAT_BPEL/webapps ; do
 #	pushd $dir > /dev/null
 #	rm -rf docs examples manager host-manager
 #	popd > /dev/null

--- a/setup.sh
+++ b/setup.sh
@@ -157,8 +157,8 @@ cat create_storedproc.sql | sudo -u postgres psql $DATABASE
 # We need to generate this on the fly since it include the user
 # defined ROLENAME.
 log "Changing owner of the stored procedure."
-echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
-cat /tmp/alter.sql | sudo -u postgres psql $DATABASE
+echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > alter.sql
+cat alter.sql | sudo -u postgres psql $DATABASE
 
 log "Securing the Tomcat installations"
 for dir in $MANAGER $BPEL ; do

--- a/setup.sh
+++ b/setup.sh
@@ -19,7 +19,8 @@ function usage()
 }
 
 function log {
-	echo $1 | tee /var/log/service-manager-install.log >&2
+	echo $1
+	echo "$(date '$b %d %Y %Y')  - $1" >> /var/log/service-manager-install.log 
 }
 
 function start_tomcat {
@@ -190,6 +191,7 @@ mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
 #done
 
 if [[ $OS = redhat7 || $OS = centos ]] ; then
+	echo "Opening port 8080"
 	iptables -I INPUT -p tcp -m tcp --dport 8080 -j ACCEPT
 fi
 

--- a/setup.sh
+++ b/setup.sh
@@ -39,7 +39,7 @@ function stop_tomcat {
 }
 
 function wait_for {
-	while ! grep "Server startup in" $1/log/catalina.out ; do
+	while ! grep "Server startup in" $1/logs/catalina.out ; do
 		echo "Waiting for $1 to start"
 		sleep 1
 	done

--- a/setup.sh
+++ b/setup.sh
@@ -139,6 +139,11 @@ cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
 cp active-bpel.xml $BPEL/conf/Catalina/localhost
 cp langrid.ae.properties $BPEL/bpr
 
+# Get the new .war file before starting Tomcat for the first time.
+log "Downloading the latest service manager war file."
+wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
+mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
+
 toggle_tomcat
 
 log "Creating indices."
@@ -170,10 +175,6 @@ for dir in $MANAGER $BPEL ; do
 	chmod -R 0770 webapps
 	popd > /dev/null
 done
-
-log "Downloading the latest service manager war file."
-wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
-mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
 
 #log "Removing default webapps."
 #for dir in $MANAGER/webapps $BPEL/webapps ; do

--- a/setup.sh
+++ b/setup.sh
@@ -41,7 +41,7 @@ function stop_tomcat {
 function wait_for {
 	while ! grep "Server startup in" $1/logs/catalina.out ; do
 		echo "Waiting for $1 to start"
-		sleep 1
+		sleep 3
 	done
 }
 
@@ -153,13 +153,14 @@ wget $manager/database-setup.sh
 #rm /tmp/*.sql
 
 #createlang plpgsql $DATABASE
-sudo -u postgres psql $DATABASE < /tmp/create_indices.sql
-sudo -u postgres psql $DATABASE < /tmp/create_storedproc.sql
+cat create_indices.sql | sudo -u postgres psql $DATABASE
+cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
+
+# We need to generate this on the fly since it include the user
+# defined ROLENAME.
 echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
 #psql $DATABSE < /tmp/alter.sql
 cat /tmp/alter.sql | sudo -u postgres psql $DATABASE
-
-#psql $DATABASE -c "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME"
 
 log "Securing the Tomcat installations"
 for dir in $MANAGER $BPEL ; do
@@ -181,12 +182,12 @@ log "Downloading the latest service manager war file."
 wget https://github.com`wget -qO- https://github.com/openlangrid/langrid/releases/latest | grep --color=never \.war\" | cut -d '"' -f 2 `
 mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
 
-log "Removing default webapps."
-for dir in $MANAGER/webapps $BPEL/webapps ; do
-	pushd $dir > /dev/null
-	rm -rf docs examples manager host-manager
-	popd > /dev/null
-done
+#log "Removing default webapps."
+#for dir in $MANAGER/webapps $BPEL/webapps ; do
+#	pushd $dir > /dev/null
+#	rm -rf docs examples manager host-manager
+#	popd > /dev/null
+#done
 
 #if [[ $OS = redhat7 ]] ; then
 #	systemctl start tomcat

--- a/setup.sh
+++ b/setup.sh
@@ -20,10 +20,11 @@ function usage()
 
 function log {
 	echo $1
-	echo "$(date +'$b %d %Y %Y')  - $1" >> /var/log/service-manager-install.log 
+	echo "$(date +'%b %d %Y %Y') - $1" >> /var/log/service-manager-install.log 
 }
 
 function start_tomcat {
+	log "Staring tomcat."
 	if [[ $OS = redhat7 || $OS = centos ]] ; then
 		systemctl start tomcat
 	else
@@ -32,6 +33,7 @@ function start_tomcat {
 }
 
 function stop_tomcat {
+	log "Stopping tomcat."
 	if [[ $OS = redhat7 || $OS = centos ]] ; then
 		systemctl stop tomcat
 	else
@@ -139,17 +141,17 @@ cp langrid.ae.properties $BPEL/bpr
 
 toggle_tomcat
 
-log "Creating indices and stored procedure."
-# Grab the SQL and setup scripts.
-wget $manager/create_storedproc.sql
+log "Creating indices."
 wget $manager/create_indices.sql
-wget $manager/database-setup.sh
-
 cat create_indices.sql | sudo -u postgres psql $DATABASE
+
+log "Creating stored procedure."
+wget $manager/create_storedproc.sql
 cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
 
 # We need to generate this on the fly since it include the user
 # defined ROLENAME.
+log "Changing owner of the stored procedure."
 echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
 cat /tmp/alter.sql | sudo -u postgres psql $DATABASE
 

--- a/setup.sh
+++ b/setup.sh
@@ -189,11 +189,9 @@ mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
 #	popd > /dev/null
 #done
 
-#if [[ $OS = redhat7 ]] ; then
-#	systemctl start tomcat
-#else
-#	service tomcat start
-#fi
+if [[ $OS = redhat7 || $OS = centos ]] ; then
+	iptables -I INPUT -p tcp -m tcp --dport 8080 -j ACCEPT
+fi
 
 start_tomcat 
 

--- a/setup.sh
+++ b/setup.sh
@@ -54,6 +54,7 @@ function toggle_tomcat {
 	wait_for $MANAGER
 	wait_for $BPEL
 	stop_tomcat
+	sleep 5
 }
 
 source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)

--- a/setup.sh
+++ b/setup.sh
@@ -2,7 +2,8 @@
 set -u
 
 smg=smg-1.1.0-SNAPSHOT
-manager=http://downloads.lappsgrid.org/service-manager
+#manager=http://downloads.lappsgrid.org/service-manager
+manager=https://raw.githubusercontent.com/lappsgrid-incubator/service-manager-installation/17-centos-start
 scripts=http://downloads.lappsgrid.org/scripts
 
 # Locations that Tomcat is installed.

--- a/setup.sh
+++ b/setup.sh
@@ -23,7 +23,7 @@ function log {
 }
 
 function start_tomcat {
-	if [[ $OS = redhat7 || $OS = centos]] ; then
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
 		systemctl start tomcat
 	else
 		service tomcat start
@@ -31,7 +31,7 @@ function start_tomcat {
 }
 
 function stop_tomcat {
-	if [[ $OS = redhat7 || $OS = centos]] ; then
+	if [[ $OS = redhat7 || $OS = centos ]] ; then
 		systemctl stop tomcat
 	else
 		service tomcat stop

--- a/setup.sh
+++ b/setup.sh
@@ -5,6 +5,10 @@ smg=smg-1.1.0-SNAPSHOT
 manager=http://downloads.lappsgrid.org/service-manager
 scripts=http://downloads.lappsgrid.org/scripts
 
+# Locations that Tomcat is installed.
+MANAGER=/usr/share/tomcat/service-manager
+BPEL=/usr/share/tomcat/active-bpel
+
 function usage()
 {
 	echo
@@ -16,6 +20,36 @@ function usage()
 
 function log {
 	echo $1 | tee /var/log/service-manager-install.log >&2
+}
+
+function start_tomcat {
+	if [[ $OS = redhat7 || $OS = centos]] ; then
+		systemctl start tomcat
+	else
+		service tomcat start
+	fi
+}
+
+function stop_tomcat {
+	if [[ $OS = redhat7 || $OS = centos]] ; then
+		systemctl stop tomcat
+	else
+		service tomcat stop
+	fi
+}
+
+function wait_for {
+	while ! grep "Server startup in" $1/log/catalina.out ; do
+		echo "Waiting for $1 to start"
+		sleep 1
+	done
+}
+
+function toggle_tomcat {
+	start_tomcat
+	wait_for $MANAGER
+	wait_for $BPEL
+	stop_tomcat
 }
 
 source <(curl -sSL http://downloads.lappsgrid.org/scripts/sniff.sh)
@@ -58,6 +92,15 @@ curl -sSL $scripts/install-java.sh | bash
 log "Installing PostgreSQL"
 curl -sSL $scripts/install-postgres.sh | bash
 
+if [[ $OS = centos || $OS = redhat7 ]] ; then
+	hba=/var/lib/pgsql/9.6/data/pg_hba.conf
+	rm $hba
+	echo "local all all trust" > $hba
+	echo "host all all 127.0.0.1/32 trust" >> $hba
+	echo "host all all ::1/128 trust" >> $hba
+	systemctl restart postgresql-9.6
+fi
+
 if [ ! -e ServiceManager.config ] ; then
 	wget $manager/ServiceManager.config
 fi
@@ -76,11 +119,14 @@ chmod +x $smg/smg
 # 	langrid.ae.properties
 # 	db.config
 $smg/smg ServiceManager.config
+source db.config
+
+sudo -u postgres createuser -S -D -R $ROLENAME
+sudo -u postgres psql --command "ALTER USER $ROLENAME WITH PASSWORD '$PASSWORD'"
+sudo -u postgres createdb $DATABASE -O $ROLENAME -E 'UTF8'
 
 # Now install Tomcat and create the PostgreSQL database.
 log "Starting Tomcat installation."
-MANAGER=/usr/share/tomcat/service-manager
-BPEL=/usr/share/tomcat/active-bpel
 curl -sSL $manager/install-tomcat.sh | bash
 
 cp tomcat-users.xml $MANAGER/conf
@@ -90,19 +136,30 @@ cp tomcat-users-bpel.xml $BPEL/conf/tomcat-users.xml
 cp active-bpel.xml $BPEL/conf/Catalina/localhost
 cp langrid.ae.properties $BPEL/bpr
 
-log "Creating role, database and stored procedure."
+toggle_tomcat
+
+log "Creating indices and stored procedure."
 # Grab the SQL and setup scripts.
 wget $manager/create_storedproc.sql
 wget $manager/create_indices.sql
 wget $manager/database-setup.sh
 
 # Generate a custom database-setup.sh with the settings generated above.
-echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
-chmod +x /tmp/database-setup.sh
-mv *.sql /tmp
+#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
+#chmod +x /tmp/database-setup.sh
+#mv *.sql /tmp
 # Now run the setup script as the postgres user.
-su - postgres -c "bash /tmp/database-setup.sh"
-rm /tmp/*.sql
+#su - postgres -c "bash /tmp/database-setup.sh"
+#rm /tmp/*.sql
+
+#createlang plpgsql $DATABASE
+sudo -u postgres psql $DATABASE < /tmp/create_indices.sql
+sudo -u postgres psql $DATABASE < /tmp/create_storedproc.sql
+echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
+#psql $DATABSE < /tmp/alter.sql
+cat /tmp/alter.sql | sudo -u postgres psql $DATABASE
+
+#psql $DATABASE -c "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME"
 
 log "Securing the Tomcat installations"
 for dir in $MANAGER $BPEL ; do
@@ -131,11 +188,13 @@ for dir in $MANAGER/webapps $BPEL/webapps ; do
 	popd > /dev/null
 done
 
-if [[ $OS = redhat7 ]] ; then
-	systemctl start tomcat
-else
-	service tomcat start
-fi
+#if [[ $OS = redhat7 ]] ; then
+#	systemctl start tomcat
+#else
+#	service tomcat start
+#fi
+
+start_tomcat 
 
 log "The Service Grid is now running."
 echo

--- a/setup.sh
+++ b/setup.sh
@@ -20,7 +20,7 @@ function usage()
 
 function log {
 	echo $1
-	echo "$(date '$b %d %Y %Y')  - $1" >> /var/log/service-manager-install.log 
+	echo "$(date +'$b %d %Y %Y')  - $1" >> /var/log/service-manager-install.log 
 }
 
 function start_tomcat {

--- a/setup.sh
+++ b/setup.sh
@@ -67,7 +67,7 @@ fi
 if [[ $OS = ubuntu ]] ; then
 	log "Updating apt-get indices."
 	apt-get update && apt-get upgrade -y
-elif [[ $OS = redhat || $OS = centos ]] ; then
+elif [[ $OS = redhat* || $OS = centos ]] ; then
 	yum makecache fast
 else
 	log "Unsupport Linux flavor: $OS"

--- a/setup.sh
+++ b/setup.sh
@@ -41,7 +41,7 @@ function stop_tomcat {
 
 function wait_for {
 	while ! grep "Server startup in" $1/logs/catalina.out ; do
-		echo "Waiting for $1 to start"
+		log "Waiting for $1 to start"
 		sleep 3
 	done
 }
@@ -145,22 +145,12 @@ wget $manager/create_storedproc.sql
 wget $manager/create_indices.sql
 wget $manager/database-setup.sh
 
-# Generate a custom database-setup.sh with the settings generated above.
-#echo '#!/usr/bin/env bash' | cat - db.config database-setup.sh > /tmp/database-setup.sh
-#chmod +x /tmp/database-setup.sh
-#mv *.sql /tmp
-# Now run the setup script as the postgres user.
-#su - postgres -c "bash /tmp/database-setup.sh"
-#rm /tmp/*.sql
-
-#createlang plpgsql $DATABASE
 cat create_indices.sql | sudo -u postgres psql $DATABASE
 cat create_storedproc.sql | sudo -u postgres psql $DATABASE 
 
 # We need to generate this on the fly since it include the user
 # defined ROLENAME.
 echo "ALTER FUNCTION \"AccessStat.increment\"(character varying, character varying, character varying, character varying, character varying, timestamp without time zone, timestamp without time zone, integer, timestamp without time zone, integer, timestamp without time zone, integer, integer, integer, integer) OWNER TO $ROLENAME" > /tmp/alter.sql
-#psql $DATABSE < /tmp/alter.sql
 cat /tmp/alter.sql | sudo -u postgres psql $DATABASE
 
 log "Securing the Tomcat installations"
@@ -191,7 +181,7 @@ mv `ls *.war | head -1` $MANAGER/webapps/service_manager.war
 #done
 
 if [[ $OS = redhat7 || $OS = centos ]] ; then
-	echo "Opening port 8080"
+	log "Opening port 8080"
 	iptables -I INPUT -p tcp -m tcp --dport 8080 -j ACCEPT
 fi
 

--- a/setup.sh
+++ b/setup.sh
@@ -39,10 +39,14 @@ function stop_tomcat {
 		systemctl stop tomcat
 	else
 		service tomcat stop
+		while [[ `ps aux | grep bootstrap.jar | grep -v grep | wc -l` -gt 0 ]] ; do
+			echo "Waiting for tomcat to shut down."
+			sleep 5
+		done
 	fi
 }
 
-function wait_for {
+function wait_for_start {
 	while ! grep "Server startup in" $1/logs/catalina.out ; do
 		log "Waiting for $1 to start"
 		sleep 3
@@ -51,8 +55,8 @@ function wait_for {
 
 function toggle_tomcat {
 	start_tomcat
-	wait_for $TOMCAT_MANAGER
-	wait_for $TOMCAT_BPEL
+	wait_for_start $TOMCAT_MANAGER
+	wait_for_start $TOMCAT_BPEL
 	stop_tomcat
 	sleep 5
 }

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -1,77 +1,66 @@
 #!/usr/bin/env bash
 
+### BEGIN INIT INFO
+# Provides:        tomcat7
+# Required-Start:  $network
+# Required-Stop:   $network
+# Default-Start:   2 3 4 5
+# Default-Stop:    0 1 6
+# Short-Description: Start/Stop Tomcat server
+### END INIT INFO
+
 TOMCAT=/usr/share/tomcat
 MANAGER=$TOMCAT/service-manager
 BPEL=$TOMCAT/active-bpel
 
-	if [[ -z $JAVA_HOME ]] ; then
+if [[ -z $JAVA_HOME ]] ; then
 	export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
 fi
 
-function is_running()
+function getpid()
 {
-    sudo ps aux | grep java | grep $1 > /dev/null
-    return $?
-}
-
-function if_running()
-{
-    program=$1
-    name=$2
-    is_running $program
-    if [ $? -eq 0 ] ; then
-		echo "Stopping $name"
-		sudo $program/shutdown.sh
-	else
-		echo "$name is already offline"
-	fi
-}
-
-function if_not_running()
-{
-    program=$1
-    name=$2
-    is_running $program
-    if [ $? -ne 0 ] ; then
-		echo "Starting $name"
-		sudo $program/startup.sh
-	else
-		echo "$name is already online."
-	fi
+    echo `ps aux | grep $1 | grep -v grep | awk '{print $2}' > /dev/
 }
 
 function start()
 {
-    if_not_running $MANAGER/bin "Service Manager"
-    sleep 1
-    if_not_running $BPEL/bin "Active BPEL"
-    sleep 1
+	for server in service-manager active-bpel ; do
+		local pid = $(get_pid $server)
+		if [[ -z $pid ]] ; then
+			$TOMCAT/$server/bin/startup.sh
+		else
+			echo "$1 is already running."
+		fi
+	done
 }
 
 function stop()
 {
-    if_running $MANAGER/bin "Service Manager"
-    sleep 1
-    if_running $BPEL/bin "Active BPEL"
-    sleep 1
+	for server in service-manager active-bpel ; do
+		local pid = $(get_pid $server)
+		if [[ -n $pid ]] ; then
+			$TOMCAT/$server/bin/shutdown.sh
+		else
+			echo "$1 is already running."
+		fi
+	done
 }
 
 function status()
 {
-    is_running $MANAGER/bin
-    if [ $? -eq 0 ] ; then
-		echo "Service Manager: online"
+	local pid=$(getpid service-manager)
+	if [[ -n $pid ]] ; then
+		echo "Service Manager: $pid"
 	else
 		echo "Service Manager: offline"
     fi
     
-    is_running $BPEL/bin 
-    if [ $? -eq 0 ] ; then
-		echo "BPEL Server    : online"
+	pid=$(getpid active-bpel)
+	if [[ -n $pid ]] ; then
+		echo "BPEL Server: $pid"
 	else
-		echo "BPEL Server    : offline"
+		echo "BPEL Server: offline"
     fi
-    echo
 }
 
 function usage()
@@ -85,16 +74,13 @@ function usage()
 function kill_tomcat()
 {
     echo "Killing zombie tomcat instances."
-    for proc in `ps a | grep java | grep ServiceGrid | cut -d\  -f1` ; do
-		echo "Killing zombie process $proc"
-		sudo kill -9 $proc
-	done
+    ps aux | grep tomcat | grep -v grep | awk '{print $2}' | xargs kill -9
 }
 
 case $1 in
     start)
 		echo "Starting the Service Grid."
-		start
+		start 
 		echo "Done"
 		;;
     stop)
@@ -107,7 +93,7 @@ case $1 in
 		;;
     restart)
 		echo "Restarting the Service Grid."
-		stop
+		stop 
 		sleep 5
 		start
 		echo "Done"

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -6,7 +6,8 @@
 ### END INIT INFO
 
 if [ -z "$JAVA_HOME" ] ; then
-	JAVA_HOME=/usr/lib/jvm/java-7-oracle
+	#JAVA_HOME=/usr/lib/jvm/java-7-oracle
+	JAVA_HOME=java-8-openjdk-amd64
 fi
 
 TOMCAT_ROOT=/usr/share/tomcat

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -4,7 +4,7 @@ TOMCAT=/usr/share/tomcat
 MANAGER=$TOMCAT/service-manager
 BPEL=$TOMCAT/active-bpel
 
-if [[ -z $JAVA_HOME ]] ; then
+	if [[ -z $JAVA_HOME ]] ; then
 	export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
 fi
 

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -19,17 +19,17 @@ fi
 
 function getpid()
 {
-    echo `ps aux | grep $1 | grep -v grep | awk '{print $2}' > /dev/
+    echo `ps aux | grep $1 | grep -v grep | awk '{print $2}'`
 }
 
 function start()
 {
 	for server in service-manager active-bpel ; do
-		local pid = $(get_pid $server)
+		local pid=$(getpid $server)
 		if [[ -z $pid ]] ; then
 			$TOMCAT/$server/bin/startup.sh
 		else
-			echo "$1 is already running."
+			echo "$server is already running."
 		fi
 	done
 }
@@ -37,11 +37,11 @@ function start()
 function stop()
 {
 	for server in service-manager active-bpel ; do
-		local pid = $(get_pid $server)
+		local pid=$(getpid $server)
 		if [[ -n $pid ]] ; then
 			$TOMCAT/$server/bin/shutdown.sh
 		else
-			echo "$1 is already running."
+			echo "$server is not running."
 		fi
 	done
 }
@@ -74,7 +74,7 @@ function usage()
 function kill_tomcat()
 {
     echo "Killing zombie tomcat instances."
-    ps aux | grep tomcat | grep -v grep | awk '{print $2}' | xargs kill -9
+    ps aux | grep bootstrap.jar | grep -v grep | awk '{print $2}' | xargs kill -9
 }
 
 case $1 in

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -1,60 +1,136 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-### BEGIN INIT INFO
-# Provides:    tomcat
-# Description: Control script for Tomcat. 
-### END INIT INFO
+TOMCAT=/usr/share/tomcat
+MANAGER=$TOMCAT/service-manager
+BPEL=$TOMCAT/active-bpel
 
-if [ -z "$JAVA_HOME" ] ; then
-	#JAVA_HOME=/usr/lib/jvm/java-7-oracle
-	JAVA_HOME=java-8-openjdk-amd64
+if [[ -z $JAVA_HOME ]] ; then
+	export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
 fi
 
-TOMCAT_ROOT=/usr/share/tomcat
-MANAGER=$TOMCAT_ROOT/service-manager
-BPEL=$TOMCAT_ROOT/active-bpel
-TOMCAT_USER=tomcat
+function is_running()
+{
+    sudo ps aux | grep java | grep $1 > /dev/null
+    return $?
+}
+
+function if_running()
+{
+    program=$1
+    name=$2
+    is_running $program
+    if [ $? -eq 0 ] ; then
+		echo "Stopping $name"
+		sudo $program/shutdown.sh
+	else
+		echo "$name is already offline"
+	fi
+}
+
+function if_not_running()
+{
+    program=$1
+    name=$2
+    is_running $program
+    if [ $? -ne 0 ] ; then
+		echo "Starting $name"
+		sudo $program/startup.sh
+	else
+		echo "$name is already online."
+	fi
+}
 
 function start()
 {
-	sudo -u $TOMCAT_USER $MANAGER/bin/startup.sh
-	sudo -u $TOMCAT_USER $BPEL/bin/startup.sh
+    if_not_running $MANAGER/bin "Service Manager"
+    sleep 1
+    if_not_running $BPEL/bin "Active BPEL"
+    sleep 1
 }
 
 function stop()
 {
-	sudo -u $TOMCAT_USER $MANAGER/bin/shutdown.sh
-	sudo -u $TOMCAT_USER $BPEL/bin/shutdown.sh
+    if_running $MANAGER/bin "Service Manager"
+    sleep 1
+    if_running $BPEL/bin "Active BPEL"
+    sleep 1
+}
+
+function status()
+{
+    is_running $MANAGER/bin
+    if [ $? -eq 0 ] ; then
+		echo "Service Manager: online"
+	else
+		echo "Service Manager: offline"
+    fi
+    
+    is_running $BPEL/bin 
+    if [ $? -eq 0 ] ; then
+		echo "BPEL Server    : online"
+	else
+		echo "BPEL Server    : offline"
+    fi
+    echo
+}
+
+function usage()
+{
+    echo
+    echo "USAGE" 
+    echo "    tomcat [start|stop|status|restart|force-stop|force-restart"
+    echo
+}
+
+function kill_tomcat()
+{
+    echo "Killing zombie tomcat instances."
+    for proc in `ps a | grep java | grep ServiceGrid | cut -d\  -f1` ; do
+		echo "Killing zombie process $proc"
+		sudo kill -9 $proc
+	done
 }
 
 case $1 in
-	start)
-		echo -n "Staring Tomcat instances"
+    start)
+		echo "Starting the Service Grid."
 		start
-		echo "Done."
+		echo "Done"
 		;;
-	stop)
-		echo "Stopping Tomcat instances"
+    stop)
+		echo "Stopping the Service Grid."
 		stop
-		echo "Done."
+		echo "Done"
 		;;
-	restart)
-		echo "Restarting Tomcat instances"
+    status)
+		status
+		;;
+    restart)
+		echo "Restarting the Service Grid."
 		stop
 		sleep 5
 		start
 		echo "Done"
 		;;
-	force-reload)
-		echo "Forcing Tomcat instances to restart."
+    force-stop)
+		echo "Killing all Tomcat instances."
+		stop
+		echo "Waiting for tomcat to quit cleanly."
+		sleep 5
+		kill_tomcat 
+		echo "Done."
+		;;
+    force-restart)
+		echo "Forcing a restart."
 		stop
 		sleep 5
-		ps a | grep tomcat | grep java | cut -d\  -f2 | xargs kill -9
-		sleep 4
+		kill_tomcat
+		sleep 2
 		start
 		echo "Done."
 		;;
-	status)
-		echo "Unsupported option: status"
+    *)
+		echo "Unknown command $1"
+		usage
 		;;
 esac

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -27,7 +27,7 @@ function start()
 	for server in service-manager active-bpel ; do
 		local pid=$(getpid $server)
 		if [[ -z $pid ]] ; then
-			$TOMCAT/$server/bin/startup.sh
+			sudo -u tomcat $TOMCAT/$server/bin/startup.sh
 		else
 			echo "$server is already running: $pid"
 		fi
@@ -39,7 +39,7 @@ function stop()
 	for server in service-manager active-bpel ; do
 		local pid=$(getpid $server)
 		if [[ -n $pid ]] ; then
-			$TOMCAT/$server/bin/shutdown.sh
+			sudo -u tomcat $TOMCAT/$server/bin/shutdown.sh
 		else
 			echo "$server is not running."
 		fi

--- a/tomcat.sh
+++ b/tomcat.sh
@@ -29,7 +29,7 @@ function start()
 		if [[ -z $pid ]] ; then
 			$TOMCAT/$server/bin/startup.sh
 		else
-			echo "$server is already running."
+			echo "$server is already running: $pid"
 		fi
 	done
 }

